homeassistant/components/shelly/event.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+from typing import Any, Final
+
+from homeassistant.components.event import (
+    DOMAIN as EVENT_DOMAIN,
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import RPC_INPUTS_EVENTS_TYPES
+from .coordinator import ShellyRpcCoordinator, get_entry_data
+from .utils import (
+    async_remove_shelly_entity,
+    get_device_entry_gen,
+    get_rpc_input_name,
+    get_rpc_key_instances,
+    is_rpc_momentary_input,
+)
+
+
+@dataclass
+class ShellyEventDescription(EventEntityDescription):
+    """Class to describe Shelly event."""
+
+    removal_condition: Callable[[dict, dict, str], bool] | None = None
+
+
+RPC_EVENT: Final = ShellyEventDescription(
+    key="input",
+    device_class=EventDeviceClass.BUTTON,
+    event_types=list(RPC_INPUTS_EVENTS_TYPES),
+    removal_condition=lambda config, status, key: not is_rpc_momentary_input(
+        config, status, key
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors for device."""
+    if get_device_entry_gen(config_entry) == 2:
+        coordinator = get_entry_data(hass)[config_entry.entry_id].rpc
+        assert coordinator
+
+        entities = []
+        key_instances = get_rpc_key_instances(coordinator.device.status, RPC_EVENT.key)
+
+        for key in key_instances:
+            if RPC_EVENT.removal_condition and RPC_EVENT.removal_condition(
+                coordinator.device.config, coordinator.device.status, key
+            ):
+                unique_id = f"{coordinator.mac}-{key}"
+                async_remove_shelly_entity(hass, EVENT_DOMAIN, unique_id)
+            else:
+                entities.append(ShellyRpcEvent(coordinator, key, RPC_EVENT))
+
+        async_add_entities(entities)
+
+
+class ShellyRpcEvent(CoordinatorEntity[ShellyRpcCoordinator], EventEntity):
+    """Represent RPC event entity."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+    entity_description: ShellyEventDescription
+
+    def __init__(
+        self,
+        coordinator: ShellyRpcCoordinator,
+        key: str,
+        description: ShellyEventDescription,
+    ) -> None:
+        """Initialize Shelly entity."""
+        super().__init__(coordinator)
+        self.input_index = int(key.split(":")[-1])
+        self._attr_device_info = {
+            "connections": {(CONNECTION_NETWORK_MAC, coordinator.mac)}
+        }
+        self._attr_unique_id = f"{coordinator.mac}-{key}"
+        self._attr_name = get_rpc_input_name(coordinator.device, key)
```

Comment:
This was changed in a later commit to remove the device and I don't agree with this change as it makes this entity behave differently from all other platforms